File: XJYChartDemo/XJYChartDemo/View/SingleLineChartTableViewCell.m
//
//  SingleLineChartTableViewCell.m
//  RecordLife
//
//  Created by JunyiXie on 2017/11/25.
//  Copyright © 2017年 谢俊逸. All rights reserved.
//

#import "SingleLineChartTableViewCell.h"

#import <XJYChart/XChart.h>

@implementation SingleLineChartTableViewCell

- (void)awakeFromNib {
  [super awakeFromNib];
  // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
  [super setSelected:selected animated:animated];

  // Configure the view for the selected state
}
- (instancetype)initWithStyle:(UITableViewCellStyle)style
              reuseIdentifier:(NSString*)reuseIdentifier {
  if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
    // XJYChart init Design for muti line
    // So @[oneitem] draw one line

    
    NSMutableArray* numberArray = @[@120,@80,@160,@120,@150];
    XLineChartItem* item =
        [[XLineChartItem alloc] initWithDataNumberArray:numberArray
                                                  color:[UIColor salmonColor]];
    XNormalLineChartConfiguration* configuration =
        [[XNormalLineChartConfiguration alloc] init];
    configuration.isShowShadow = YES;
    XLineChart* lineChart =
        [[XLineChart alloc] initWithFrame:CGRectMake(0, 0, 375, 200)
                            dataItemArray:[NSMutableArray arrayWithObject:item]
                        dataDiscribeArray:[NSMutableArray arrayWithArray:@[
                          @"January", @"February", @"March", @"April", @"May"
                        ]]
                                topNumber:@240
                             bottomNumber:@0
                                graphMode:MutiLineGraph
                       chartConfiguration:configuration];

    [self.contentView addSubview:lineChart];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
  }
  return self;
}

@end

File: XJYChartDemo/XJYChartDemo-Swift/View/SingleLineChartTableViewCell.swift
//
//  SingleLineChartTableViewCell.swift
//  XJYChartDemo-Swift
//
//  Created by Kelly Roach on 8/17/18.
//  Copyright © 2018 JunyiXie. All rights reserved.
//

import UIKit
import XJYChart

class SingleLineChartTableViewCell: UITableViewCell {
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        // Configure the view for the selected state
    }

    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

            // XJYChart init Design for muti line
            // So @[oneitem] draw one line
        let numberArray = [120, 80, 160, 120, 150]
        let item = XLineChartItem(dataNumber: NSMutableArray(array: numberArray), color: UIColor.salmon())
        let configuration = XNormalLineChartConfiguration()
        configuration.isShowShadow = false
            //    configuration.isEnableNumberLabel = YES;
        let lineChart = XLineChart(frame: CGRect(x: 0, y: 0, width: 375, height: 200), dataItemArray: [item as Any], dataDiscribeArray: ["January", "February", "March", "April", "May"], topNumber: 240, bottomNumber: 0, graphMode: XLineGraphMode.MutiLineGraph, chartConfiguration: configuration)
        contentView.addSubview(lineChart!)
        selectionStyle = UITableViewCellSelectionStyle.none
    
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
